### PR TITLE
[Onprem] Can launch Sky onprem when there are no public clouds on laptop

### LINKED ISSUE #1528: [Onprem] `sky check` with no clouds (only local) will error
Raising error might be an intended behavior, but I think, based on initial user feedback, we should just log to user that no clouds are enabled instead of raising error.

LOC indicated here: https://github.com/skypilot-org/skypilot/blob/master/sky/backends/backend_utils.py#L2106

### PR DESCRIPTION
**Pull request by michaelzhiluo**
Fixes #1528. Prior to this PR, when running `sky launch -c my-local-cluster -- ''` on a completely new machine with no public clouds:
```
(base) ubuntu@ip-172-31-47-168:~/skypilot/sky$ sky launch -c asdf -- 'echo hi'
Task from command: echo hi
RuntimeError: Cloud access is not set up. Run: sky check
```

Now it runs properly.

Tested (run the relevant ones):

- [x] Any manual or new tests for this the PR
- [ ] All smoke tests: `bash tests/run_smoke_tests.sh` 
- [x] Relevant individual smoke tests: `bash tests/run_smoke_tests.sh test_fill_in_the_name` 
- [x] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/optimizer.py [hunk 1]
```diff
@@ -826,7 +825,6 @@ def _fill_in_launchable_resources(
     try_fix_with_sky_check: bool = True,
 ) -> Tuple[Dict[resources_lib.Resources, List[resources_lib.Resources]],
            _PerCloudCandidates]:
-    backend_utils.check_public_cloud_enabled()
```
Comment (michaelzhiluo) [inline sky/optimizer.py:829]: @Michaelvll Do you remember the context when this line of code is added? Correct me if I'm wrong, but `backend_utils.check_public_cloud_enabled()` seems to be redundant as it is checked in `cli.py`.

Comment (romilbhardwaj) [reply sky/optimizer.py:829]: Perhaps it was added for the programmatic API?

File: sky/cli.py [hunk 2]
```diff
@@ -683,7 +683,8 @@ def _launch_with_confirm(
     if maybe_status is None:
         # Show the optimize log before the prompt if the cluster does not exist.
         try:
-            backend_utils.check_public_cloud_enabled()
+            if not is_local_cloud:
+                backend_utils.check_public_cloud_enabled()
```
Comment (romilbhardwaj) [inline sky/cli.py:687]: Curious - why don't we check only for public clouds and not all clouds? More specifically, why isn't this method just `backend_utils.check_cloud_enabled()`, which won't raise a runtime error when only local cloud is enabled? 

Could we change the definition of `enable` for `local` cloud to mean when one or more valid local cluster yaml is present in `~/.sky/local`?

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @michaelzhiluo! Left a comment

**Comment by Vinno97** [comment]
Is there any update on this PR? If there's a specific thing that is still blocking? I'm not familiar with the specific codebase, but would be happy to help get this resolved.

**Pull request closed without merging**